data/libs/FlightLog.lua

```diff
@@ -344,30 +344,39 @@ FlightLog = {
 		text = text or ""
 		local location = ""
 		local state = Game.player:GetFlightState()
+		local path = ""
 
 		if state == "DOCKED" then
 			local station = Game.player:GetDockedWith()
 			local parent_body = station.path:GetSystemBody().parent.name
 			location = {station.type, station.label, parent_body}
+			path = Game.system.path
 		elseif state == "DOCKING" or state == "UNDOCKING" then
 			location = {state, Game.player:FindNearestTo("SPACESTATION").label}
+			path = Game.system.path
 		elseif state == "FLYING" then
-			location = {state, Game.player.frameBody.label}
+			if Game.player.frameBody then
+				location = {state, Game.player.frameBody.label}
+			else
+				location = {state, Game.system.name} -- if orbiting a system barycenter, there will be no frame object
+			end
+			path = Game.system.path
 		elseif state == "LANDED" then
+			path = Game.system.path
 			local alt, vspd, lat, long = Game.player:GetGPS()
 			if not (lat and long) then
 				lat, long = "nil", "nil"
 			end
 			location = {state, Game.player:FindNearestTo("PLANET").label, lat, long}
 		elseif state == "JUMPING" or state == "HYPERSPACE" then
-			-- bug: if in hyperspace, there's no Game.system (!)
-			-- local spath, sysname = Game.player:GetHyperspaceDestination()
-			sysname = ""
+			--if in hyperspace, there's no Game.system
+			local spath, sysname = Game.player:GetHyperspaceDestination()
+			path = spath
 			location = {state, sysname}
```

Comment:
Is the `spath` not the same type as the `Game.system.path` or why is the version bump needed?

(I would expect the bump to be needed if unpacking the table is different, but then I'd expect to see edits in the pigui module, that makes use of it, as well).

Reply:
I thought the same way (and I still think the same way), but I couldn't find any explanation for the Flight Log getting erased. I read the whole file over and over again but couldn't find what was wrong, and only bumping up the version number solved it. It still bugs me.

Reply:
I see you pushed again, but not what the change was (just a rebase?). Regarding `location`, looks like it should be a tuple (if indeed that be its lua name), of two strings, so if `sysname` is still a string, that should be fine, I suspect.

I'd merge this in a second if I could understand why the flight log save bump is needed (me not understanding it hints something is wrong in either my brain or the code (although one does not exclude the other)). Granted, I haven't banged my head against a brick wall to understand that particular thingy, yet.

Reply:
Read my last PR comment, resolved.